app/controllers/subject_plans_controller.rb

```diff
@@ -46,10 +46,20 @@ def set_planned_and_not_planned_subjects
     @planned_subjects =
       TreePreloader.preload(current_user.planned_subjects.select('subjects.*', 'subject_plans.semester')
         .order('subject_plans.semester'))
-    @not_planned_approved_subjects, @not_planned_subjects =
-      TreePreloader.preload(Subject.ordered_by_category.ordered_by_short_or_full_name).reject { |subject|
-        current_user.planned?(subject)
-      }.partition { |subject| current_student.approved?(subject) }
+    @not_planned_approved_subjects = TreePreloader.preload(not_planned_approved_subjects)
+    @not_planned_subjects = TreePreloader.preload(not_planned_subjects)
+  end
+
+  def not_planned_subjects
+    current_degree
+      .subjects
+      .where.not(id: current_user.planned_subjects)
+      .ordered_by_category
+      .ordered_by_short_or_full_name
+  end
+
+  def not_planned_approved_subjects
+    not_planned_subjects.where(id: current_student.approved_subjects)
```

Comment:
Qué te parece hacer:


```suggestion
    not_planned_subjects.select { |subject| current_student.approved?(subject) }
```

para no tener que hacer otra query?

Reply:
De todas formas esto va a cambiar en #877... capaz que podemos mover estos cambios a ese PR?

Reply:
De última podemos hacer que este PR solo agregue el scope para que después se use en el otro PR...